src/components/Announcements/AnnouncementAction/AnnouncementAction.test.tsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import { ThemeProvider } from '@emotion/react';
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import TestRouter from '__tests__/util/TestRouter';
+import { ActionStyleEnum } from 'src/graphql/types.generated';
+import theme from 'src/theme';
+import { ActionFragment } from '../Announcements.generated';
+import { AnnouncementAction } from './AnnouncementAction';
+
+const handlePerformAction = jest.fn();
+const accountListId = 'accountListId';
+const router = {
+  query: { accountListId },
+  isReady: true,
+};
+const defaultTextAndIconColor = '#EEEEEE';
+
+interface AnnouncementActionProps {
+  actionStyle?: ActionStyleEnum;
+  textAndIconColor?: string;
+  isBanner?: boolean;
+}
+const TestComponent: React.FC<AnnouncementActionProps> = ({
+  actionStyle = ActionStyleEnum.Link,
+  textAndIconColor = defaultTextAndIconColor,
+  isBanner = false,
+}) => {
+  const action: ActionFragment = {
+    id: 'action-1',
+    style: actionStyle,
+    label: 'Contacts',
+    args: 'coaching-account',
+  };
+
+  return (
+    <TestRouter router={router}>
+      <ThemeProvider theme={theme}>
+        <AnnouncementAction
+          action={action}
+          textAndIconColor={textAndIconColor}
+          handlePerformAction={handlePerformAction}
+          isBanner={isBanner}
+        />
+      </ThemeProvider>
+    </TestRouter>
+  );
+};
+
+describe('AnnouncementAction', () => {
+  describe('Icon', () => {
+    it('initial', () => {
+      const { getByRole, queryByText } = render(
+        <TestComponent actionStyle={ActionStyleEnum.Icon} />,
+      );
+
+      expect(queryByText('Contacts')).not.toBeInTheDocument();
+
+      const icon = getByRole('button', { name: 'Contacts' });
+      userEvent.click(icon);
+      expect(handlePerformAction).toHaveBeenCalled();
+    });
+  });
+  describe('Button', () => {
+    it('initial', () => {
+      const { getByText } = render(<TestComponent />);
+
+      const button = getByText('Contacts');
+      expect(button).toBeInTheDocument();
+      userEvent.click(button);
```

Comment:
`getByText` implicitly asserts the element's existence.

```suggestion
      userEvent.click(getByText('Contacts'));
```